apps/dhparam.c

```diff
@@ -242,10 +242,19 @@ int dhparam_main(int argc, char **argv)
         } else
 # endif
         {
-            if (informat == FORMAT_ASN1)
+            if (informat == FORMAT_ASN1) {
+                /*
+                 * We have no PEM header to determine what type of DH params it
```

Comment:
I'm not a fan of "magic try both" kinds of things.  Is that better than requiring  the user to explicitly say what it is?